File: verification/council/provider.go
package council

import (
	"context"
)

type CouncilVerificationResult struct {
	IsVerified          bool   `json:"is_verified"`
	DoctorName          string `json:"doctor_name"`
	RegistrationNumber  string `json:"registration_number"`
	CouncilName         string `json:"council_name"`
	RegistrationYear    int    `json:"registration_year"`
	Status              string `json:"status"` // "ACTIVE", "SUSPENDED", "NOT_FOUND", "MANUAL_REVIEW_REQUIRED"
	VerificationSource  string `json:"verification_source"`
	Remarks             string `json:"remarks,omitempty"`
}

type CouncilVerificationProvider interface {
	Verify(ctx context.Context, registrationNumber string, council string) (*CouncilVerificationResult, error)
	GetProviderName() string
}

File: backend_api/verification/council/provider.py
from abc import ABC, abstractmethod
from typing import Optional

class CouncilVerificationResult:
    def __init__(
        self,
        is_verified: bool,
        status: str,
        verification_source: str,
        doctor_name: str = "",
        registration_number: str = "",
        council_name: str = "",
        registration_year: int = 0,
        remarks: str = ""
    ):
        self.is_verified = is_verified
        self.doctor_name = doctor_name
        self.registration_number = registration_number
        self.council_name = council_name
        self.registration_year = registration_year
        self.status = status
        self.verification_source = verification_source
        self.remarks = remarks

class CouncilVerificationProvider(ABC):
    @abstractmethod
    def verify(self, ctx: dict, registration_number: str, council: str) -> CouncilVerificationResult:
        pass

    @abstractmethod
    def get_provider_name(self) -> str:
        pass
